src/composables/transactionList.ts

```diff
@@ -0,0 +1,324 @@
+  return transactions.value[address]?.loaded.find((transaction) => transaction.hash === hash);
+}
+
+function updateAccountTransactions(address: Encoded.AccountAddress, transaction: ITransaction) {
+  const transactionIndex = transactions.value[address]?.loaded?.findIndex(
+    (tr) => tr.hash === transaction.hash,
+  );
+
+  if (transactionIndex >= 0) {
+    transactions.value[address].loaded[transactionIndex] = transaction;
+  }
+}
+
+const setTransactionsNextPage = (address: Encoded.AccountAddress, url: string) => {
+  transactions.value[address].nextPageUrl = url;
+};
+
+export const useTransactionList = () => {
+  const store = useStore();
+  const { nodeNetworkId, getAeSdk } = useAeSdk({ store });
+  const { activeAccount, isLoggedIn } = useAccounts({ store });
+
+  const allTransactionsForAccount = computed(() => {
+    if (!isLoggedIn) {
+      return [];
+    }
+    const { pending, loaded } = getOrInitTransactionStateForAccount(activeAccount.value.address);
+
+    return [...loaded, ...(pending[nodeNetworkId.value!] || [])];
+  });
+
+  function setPendingTransactionSentByHash(address:Encoded.AccountAddress, hash: string) {
+    const index = transactions.value[address].pending[nodeNetworkId.value!]
+      ?.findIndex((transaction) => transaction.hash === hash);
+    if (index >= 0) {
+      transactions.value[address]
+        .pending[nodeNetworkId.value!][index][TX_DIRECTION.sent] = true;
+    }
+  }
+
+  async function waitTransactionMined(address: Encoded.AccountAddress, hash: string) {
+    try {
+      const aeSdk = await getAeSdk();
+      await aeSdk.poll(hash as Encoded.TxHash);
+      setPendingTransactionSentByHash(address, hash);
+    } catch (e) {
+      store.commit('removePendingTransactionByHash', { hash, network: nodeNetworkId.value });
+    }
+  }
+
+  function addCustomPendingTransactionForAccount(
+    address: Encoded.AccountAddress,
+    transaction: ITransaction,
+  ) {
+    if (!transactions.value[address]) {
+      transactions.value[address] = generateEmptyTransactionState();
+    }
+    transactions.value[address].pending[nodeNetworkId.value!] = [transaction];
+    waitTransactionMined(address, transaction.hash);
+  }
+
+  async function fetchTransactionsFromMiddleware(
+    transactionState: ITransactionsState,
+    address: Encoded.AccountAddress,
+    recent: boolean,
+    limit: number,
+  ) {
+    const { fetchFromMiddlewareCamelCased } = useMiddleware({ store });
+
+    const url = (recent || transactionState.nextPageUrl === '')
+      ? `/v2/accounts/${address}/activities?limit=${limit}`
+      : transactionState.nextPageUrl!;
+
+    try {
+      const { data, next } = await fetchFromMiddlewareCamelCased(url);
+      if (!recent) {
+        setTransactionsNextPage(address, next);
+      }
+      return data;
+    } catch (error) {
+      return [];
+    }
+  }
+
+  async function fetchPendingTransaction(
+    address: Encoded.AccountAddress,
```

Comment:
I think this argument can be moved to the same line as the function name - same as othe functions are defined.